packages/plans-grid-next/src/components/comparison-grid/index.tsx

```diff
@@ -988,39 +993,12 @@ const ComparisonGrid = ( {
 		}
 
 		if ( newVisiblePlans.length !== visibleLength ) {
-			newVisiblePlans = newVisiblePlans.slice( 0, visibleLength );
+			newVisiblePlans = displayedGridPlans
+				.slice( 0, visibleLength )
+				.map( ( { planSlug } ) => planSlug );
+			setVisibleGridPlans( newVisiblePlans );
 		}
-
-		setVisiblePlans( newVisiblePlans );
-	}, [ gridSize, displayedGridPlans, isInSignup ] );
-
-	const visibleGridPlans = useMemo(
-		() =>
-			visiblePlans.reduce( ( acc, planSlug ) => {
-				const gridPlan = displayedGridPlans.find(
-					( gridPlan ) => getPlanClass( gridPlan.planSlug ) === getPlanClass( planSlug )
-				);
-
-				if ( gridPlan ) {
-					acc.push( gridPlan );
-				}
-
-				return acc;
-			}, [] as GridPlan[] ),
-		[ visiblePlans, displayedGridPlans ]
-	);
-
-	const onPlanChange = useCallback(
-		( currentPlan: PlanSlug, event: ChangeEvent< HTMLSelectElement > ) => {
-			const newPlan = event.currentTarget.value;
-			const newVisiblePlans = visiblePlans.map( ( plan ) =>
-				plan === currentPlan ? ( newPlan as PlanSlug ) : plan
-			);
-
-			setVisiblePlans( newVisiblePlans );
-		},
-		[ visiblePlans ]
-	);
```

Comment:
Let's see what the end result will be, but my feeling is we should move this back here, closer to the other logic above. It's a little scattered now and might work adversely towards additional refactors.